src/Main.tsx

```diff
@@ -9,63 +9,88 @@ import {
 } from '@material-ui/core'
 import {ManagedQuestionJSON} from './types'
 import queryString from 'query-string'
+import { useWindowSize } from "@reach/window-size";
 
 
 interface Props {
   managedQuestions: ManagedQuestionJSON;
   allTags: string[];
   search: string;
-  setSearch: Function;
+  allCategories: string[];
 }
 
 const Main = (props: Props) => {
   const history = useHistory()
+  const windowSize = useWindowSize()
 
   const params = queryString.parse(history.location.search)
-  if (typeof params.search !== 'string') {
-    params.search = ''
-  }
-  if (typeof params.integration !== 'string') {
-    params.integration = ''
+  if (typeof params.integrations !== 'string') {
+    params.integrations = ''
   }
   if (typeof params.tags !== 'string') {
     params.tags = ''
   }
+  if (typeof params.tagFilter !== 'string') {
+    params.tagFilter = ''
+  }
 
-  useEffect(() => {
-    console.log(params)
-    props.setSearch(params.search)
-  }, [])
-
-  const [integration, setIntegration] = useState((params.integration === '') ? 'none' : params.integration)
+  const [integrations, setIntegrations] = useState<string[]>((params.integrations !== '') ? params.integrations.split(',') : [])
   const [tags, setTags] = useState<string[]>((params.tags !== '') ? params.tags.split(',') : [])
-  const [questionNumber, setQuestionNumber] = useState<number>(10)
+  const [tagFilter, setFilterLogic] = useState<string>((params.tagFilter !== '') ? params.tagFilter : 'all')
+  const [categories, setCategories] = useState<string[]>([])
 
   useEffect(() => {
+
     const searchString : string = '/filter?'
       + ((tags.length !== 0) ? `&tags=${tags.join(',')}` : "")
-      + ((integration !== 'none') ? `&integration=${integration}` : "")
+      + ((integrations.length !== 0) ? `&integrations=${integrations.join(',')}` : "")
       + ((props.search !== '') ? `&search=${props.search}` : "")
+      + ((tagFilter !== '') ? `&tagFilter=${tagFilter}` : "")
 
     history.replace(searchString)
 
-  }, [tags, integration, props.search])
+  }, [tags, integrations, props.search, tagFilter, categories])
 
   return (
     <>
-      <Box mt={2} style={{display: 'flex'}}>
+      <Box mt={2} style={{display: windowSize.width > 750 ? 'flex' : 'block'}}>
         <Filters
-          clear={() => {
-            setIntegration('none')
-            setTags([])
-            props.setSearch('')
-          }}
           managedQuestions={props.managedQuestions}
           allTags={props.allTags}
-          integration={integration === '' ? 'none' : integration}
-          integrationClicked={setIntegration}
+          integrations={integrations}
+          integrationClicked={(integration: string) => {
+            setIntegrations((prev: string[]) => {
+              if (prev.includes(integration)) {
+                const index = prev.indexOf(integration);
```

Comment:
You've got some serious duplicate code here (see lines around line 80). Also, I don't want to see the "how" here but "what" you're trying to do. Think of how you might use a function to accomplish this.